### ISSUE TITLE
coredump on exit

### ISSUE THREAD

**Issue by aotto1968** [storaged-project/udisks#1024]
Hi, I have a lot of **udisksd** dumps on latest _opensuse_.

```
[New LWP 4027]
[New LWP 4028]
[New LWP 4030]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `/usr/libexec/udisks2/udisksd'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007fa6fcc2380c in __pthread_kill_implementation () from /lib64/libc.so.6
[Current thread is 1 (Thread 0x7fa6fc789140 (LWP 4027))]
#0  0x00007fa6fcc2380c in __pthread_kill_implementation () from /lib64/libc.so.6
#1  0x00007fa6fcbd0846 in raise () from /lib64/libc.so.6
#2  0x00007fa6fcbb981c in abort () from /lib64/libc.so.6
#3  0x00007fa6fcdaeee8 in ?? () from /lib64/libglib-2.0.so.0
#4  0x00007fa6fce0f3ba in g_assertion_message_expr () from /lib64/libglib-2.0.so.0
#5  0x00007fa6fcfce5d6 in g_static_resource_fini () from /lib64/libgio-2.0.so.0
#6  0x00007fa6fd3a8c6e in _dl_fini () from /lib64/ld-linux-x86-64.so.2
#7  0x00007fa6fcbd30c5 in __run_exit_handlers () from /lib64/libc.so.6
#8  0x00007fa6fcbd3250 in exit () from /lib64/libc.so.6
#9  0x00007fa6fcbba5b7 in __libc_start_call_main () from /lib64/libc.so.6
#10 0x00007fa6fcbba679 in __libc_start_main_impl () from /lib64/libc.so.6
#11 0x000055fe74c5cd35 in _start ()
```

```
NAME="openSUSE Tumbleweed"
# VERSION="20221003"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20221003"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:tumbleweed:20221003"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:Tumbleweed"
LOGO="distributor-logo-Tumbleweed"
```

**Comment by tbzatek** [comment]
This is #926

**Comment by xdegaye** [comment]
The crash reported on ArchLinux udisks2 version: 2.9.4-1 at [FS#76345](https://bugs.archlinux.org/task/76345) seems to be similar.
Please note that the commit [6464e30](https://github.com/storaged-project/udisks/commit/6464e3083c27b9e4d97848b9e69e862f265511d5) in PR #926 is **not included** in the release of  udisks2 2.9.4.

**Comment by tbzatek** [comment]
Distributors are free to include the menioned change in their packages. Otherwise this will be included in the upcoming 2.10.0 release (in a couple of months hopefully).
